### DISCUSSION TITLE
Using templater to pull a variable (username) once from a file

### DISCUSSION THREAD

**Discussion by cthulhubuddha** [SilentVoid13/Templater#727] [Help]
I'm not sure if this is possible, but I have a repo shared with others via obsidian git. I would like to use Templater to pull a variable Username from a common note location (say everyone has a base note called Index) across the different local repos. This would allow the template to indicate the person who originally created the note, but it should only allow it to occur once, and shouldn't change when someone else sync the repo to their Username variable

I can get this to kind of work using dataview, but that would change the variable Username to whomever's local Username is. I do not wish for the template to modify the original Username for each person's repo that pulls it down.

Is this possible with Templater today? Thanks!

**Comment by AB1908** [comment]
Certainly is but you'll likely need to use Obsidian API for this. The way I see it, there's a note elsewhere which can have front matter indicating the current user and we can pull that field into the current note. How do you envision it?

**Reply by cthulhubuddha** [reply]
i think you're envisioning it the way I am. Frontmatter in a common note with a Name field, templater pulls that Name from the common note and adds to a "Created By" field to a new note that shouldn't change to anyone else's Name in their local vault once they have sync'd the git repo. Thanks!

**Reply by AB1908** [reply]
You'll need to use `tp.file.find_tfile("[[note]]")` to grab the `TFile` first and then extract its metadata using `app.metadataCache.getFileCache(tfile)` using the TFile from earlier. That should give you the metadata for the note like its frontmatter.
